libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt

```diff
@@ -98,14 +99,25 @@ internal class RouteAlternativesController constructor(
             // TODO make async
             val alternatives: List<DirectionsRoute> = runBlocking {
                 routeAlternatives.mapIndexedNotNull { index, routeAlternative ->
-                    val alternative = parseNativeDirectionsAlternative(
-                        routeAlternative.routeResponse,
-                        routeProgress.route.routeOptions()
-                    )
-                    if (alternative == null) {
-                        logE(TAG, Message("null alternative at index $index"))
+                    try {
+                        val alternative = parseNativeDirectionsAlternative(
+                            ThreadController.IODispatcher,
+                            routeAlternative.routeResponse,
+                            routeProgress.route.routeOptions()
+                        )
+                        alternative
+                    } catch (ex: Exception) {
+                        logE(
+                            TAG,
+                            Message(
+                                """
+                                    |unable to parse alternative at index $index;
+                                    |failure for response: ${routeAlternative.routeResponse}
```

Comment:
Isn't it a huge json for Logcat? Messages are truncated, if they are longer than 4000 characters.

Reply:
This is to cover an edge case where Nav Native would pass to us an error body as successful response. We need to be able to print that whole body to know what the actual problem was. I'm kinda willing to accept the fact that if it fails for any other reason, the message with full route will be truncated.